hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -310,30 +341,67 @@ public static CryptoTransferWrapper decodeCryptoTransfer(
-                final var abiNftExchanges = (Tuple[]) tupleNested.get(2);
-                if (abiNftExchanges.length > 0) {
-                    nftExchanges = bindNftExchangesFrom(tokenType, abiNftExchanges, aliasResolver);
-                }
+    /**
+     * Decodes the given bytes of the cryptoTransfer function parameters
+     *
+     * <p><b>Important: </b>This is the latest version and supersedes public static
+     * CryptoTransferWrapper decodeCryptoTransfer(). The selector for this function is derived from:
+     * cryptoTransfer(((address,int64,bool)[]),(address,(address,int64,bool)[],(address,address,int64,bool)[])[])
+     * The first parameter describes hbar transfers and the second describes token transfers
+     *
+     * @param input encoded bytes containing selector and input parameters
+     * @param aliasResolver function used to resolve aliases
+     * @return CryptoTransferWrapper codec
+     */
+    public static CryptoTransferWrapper decodeCryptoTransferV2(
+            final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
+        final Tuple decodedTuples =
+                decodeFunctionCall(input, CRYPTO_TRANSFER_SELECTOR_V2, CRYPTO_TRANSFER_DECODER_V2);
+        List<SyntheticTxnFactory.HbarTransfer> hbarTransfers = new ArrayList<>();
+        final List<TokenTransferWrapper> tokenTransferWrappers = new ArrayList<>();
 
-                tokenTransferWrappers.add(
-                        new TokenTransferWrapper(nftExchanges, fungibleTransfers));
-            }
+        final Tuple[] hbarTransferTuples = ((Tuple) decodedTuples.get(0)).get(0);
+        final var tokenTransferTuples = decodedTuples.get(1);
+
+        if (hbarTransferTuples.length > 0) {
+            hbarTransfers = bindHBarTransfersFrom(hbarTransferTuples, aliasResolver);
         }
 
+        decodeTokenTransfer(aliasResolver, tokenTransferWrappers, (Tuple[]) tokenTransferTuples);
+
         return new CryptoTransferWrapper(new TransferWrapper(hbarTransfers), tokenTransferWrappers);
     }
 
+    public static void decodeTokenTransfer(
+            UnaryOperator<byte[]> aliasResolver,
+            List<TokenTransferWrapper> tokenTransferWrappers,
+            Tuple[] tokenTransferTuples) {
+        for (final var tupleNested : tokenTransferTuples) {
+            final var tokenType = convertAddressBytesToTokenID(tupleNested.get(0));
+
+            var nftExchanges = NO_NFT_EXCHANGES;
+            var fungibleTransfers = NO_FUNGIBLE_TRANSFERS;
+
+            final var abiAdjustments = (Tuple[]) tupleNested.get(1);
+            if (abiAdjustments.length > 0) {
+                fungibleTransfers =
+                        bindFungibleTransfersFrom(tokenType, abiAdjustments, aliasResolver);
+            }
+            final var abiNftExchanges = (Tuple[]) tupleNested.get(2);
+            if (abiNftExchanges.length > 0) {
+                nftExchanges = bindNftExchangesFrom(tokenType, abiNftExchanges, aliasResolver);
+            }
```

Comment:
Shouldn't we also alter [`bindFungibleTransferFrom` and `bindNftExchangesFrom` ](https://github.com/hashgraph/hedera-services/blob/53d7b5249d54f7878a452872a0f09da434d8b9b4/hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/codec/DecodingFacade.java#L175-L203)to take into account [the new `isApproval` field in the Solidity structs](https://github.com/hashgraph/hedera-smart-contracts/blob/7374d6ead7e5c13cd726e553d269507043241f77/contracts/hts-precompile/IHederaTokenService.sol#L46) or is that planned for a future PR?

Reply:
This planned for a future pr but thanks for pointing this out.